swap out legacy combobox for

jxjj opened the pull request:
Swaps legacy combobox for new combobox introduced in #214 

Adds a few things to the new combobox:
- a slot param for getting the `query` which we use for "Add New Option" buttons.
- optionally teleport options list to body, needed for the Leaves Table because CSS sucks. 
- configurable strategy - defaults to absolute positioning, but we can use fixed positioning if needed. (Ultimately, the teleport option resolved the Leaves Table clipping, but changing strategy to `fixed` would have also worked)
- autoupdate position on scroll

We could probably tweak some style and alignment things, as the box can be a little taller than the previous select since it shows any secondary label. Let me know if anything worth polishing up.

On dev.

cmcfadden reviewed (approved):
looks good aside from a spacing issue mentioned on slack

jxjj commented:
A few updates and fixes:

- Fixed tests. I forgot that if users can add new option, hitting enter will create the option. (The tests remembered though). We're now triggering adding new options if user is hitting enter (when permitted).
- moved the `addNewOption` functionality into the ComboBox component, so that I could handle `enter` events and combobox closing more gracefully.
- fixed an issue where the combobox doesn't always select the new option once added. This would happen if a user added an new option, and then decided to clear their query and add a different new option. This issue was caused by the highlightedOption value not resetting as the query changes.
- fixed an issue showing more than one item selected (due to no id for newly created options)
- fixed some keyboard nav issues with newly created options (again, due to no id)
- fix an issue where pressing arrow down to advance options wouldn't  keep the highlighted option in view.

On dev, if you want to poke some more.
